sympy/solvers/decompogen.py

```diff
@@ -37,17 +37,25 @@ def decompogen(f, symbol):
     if isinstance(f, (Function, Pow)):
         if f.args[0] == symbol:
             return [f]
-        result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
+        try:
+            result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
+        except TypeError:
+            result = [f]
         return result
 
     # ===== Convert to Polynomial ===== #
-    fp = Poly(f)
+    try:
+        fp = Poly(f)
+    except:
```

Comment:
using `Poly(f, symbol)` would not be a great idea. Not every time the generator would be `symbol`.
Consider, `sin(x)**2 + sin(x) + 1`, generator here is `sin(x)`.